app/controllers/tracks_controller.rb

```diff
@@ -11,6 +11,9 @@ class TracksController < ApplicationController
   def index
     authorize Track
     @tracks = apply_scopes(policy_scope(Track))
+              .joins("LEFT JOIN plays ON tracks.id = plays.track_id AND plays.user_id = #{current_user.id}")
```

Comment:
Even though `current_user.id` is not (really) user-controlled, string interpolation into raw SQL is a big no-no.

Reply:
I've recreated this with Arel

Reply:
Can't you write `.joins("LEFT JOIN plays ON tracks.id = plays.track_id AND plays.user_id = ?", current_user.id)`? Most activerecord query methods support that syntax.

Reply:
Or maybe a `.where("plays.user_id = ?", current_user.id)` after the join would be cleaner. I feel like that should give the same result?

Reply:
Your suggestions do not work. Postgres throws a weird `RuntimeError (unknown class: Integer)` without a traceback.

Where certainly does not work. A `where`-filter would only return those track that have plays.